R/io.R

```diff
@@ -297,36 +297,58 @@ reclassify_raster <- function(crop_cover_levels){
-  cdl_classes$GCAM_ID <- car::recode(cdl_classes$ID, "c(1,12,13) = 1;
-                                                      c(2,32,36) = 2;
-                                                      c(59,60) = 3;
-                                                      c(37,58,224) = 4; c(10,11,14,42,44,47,48,49,50,51,52,53,54,55,56,57,61,66,67,68,69,70,71,72,74,75,
-                                                      76,77,204,206,207,208,209,210,212,213,214,215,216,217,218,219,220,221,222,223,227,229,230,231,232,233,
-                                                      234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,225,226) = 5;
-                                                      c(5,6,26,31,33,34,35,38,211) = 6;
-                                                      c(4,21,25,27,28,29,39,205) = 7;
-                                                      3 = 9;
-                                                      c(43,46) = 10;
-                                                      c(41,45) = 11;
-                                                      c(22,23,24,30) = 12;
-                                                      c(82,121,122,123,124,63,64,65,81,83,87,88,92,
-                                                      111,112,131,141,142,143,152,176,190,195,0) = NA", as.factor = FALSE)
-
-  # Join tables by GCAM_ID and add column to seperate crops and land classes. Rename columns for simplicity.
-  reclass_table <- dplyr::left_join(cdl_classes, gcam_classes, by = "GCAM_ID") %>%
-    mutate(is_crop = if_else(is.na(GCAM_commodity), FALSE, TRUE))
-
-  reclass_table_df <- dplyr::rename(reclass_table, CDL_ID = ID,
-                                            GCAM_Class = GCAM_commodity)
+  # First link uses the GCAM crop table and links the CDL table by similar crop names. 40 IDs get filled by this operation.
+  gcam_csv %>%
+    dplyr::select(item, GTAP_crop, GCAM_commodity, IFA2002_crop) %>%
+    left_join(cdl_classes, by = c("GTAP_crop" = "CDL_Class")) %>%
+    left_join(cdl_classes, by = c("IFA2002_crop" = "CDL_Class")) %>%
+    mutate(ID = case_when(
+      is.na(ID.x) & is.na(ID.y) ~ NA_integer_,
+      is.na(ID.x) & !is.na(ID.y) ~ ID.y,
+      !is.na(ID.x) & is.na(ID.y) ~ ID.x,
+      !is.na(ID.x) & !is.na(ID.y) ~ ID.x)) %>%
+    left_join(gcam_classes, by = "GCAM_commodity") %>%
+    dplyr::select(ID, GCAM_ID, item) %>%
+    left_join(cdl_classes, gcam_csv, by = "ID") -> gcam_link
+
+  # This is then merged with the CDL IDs. All of the IDs that are still NA are made into a new table to do the second link.
+  crop_cover_levels %>%
+    left_join(gcam_link, by = "ID") %>%
+    left_join(gcam_classes, by = "GCAM_ID") %>%
+    dplyr::select(ID, Class_Names, GCAM_ID, CDL_Class, GCAM_commodity) -> crop_level_join
+
+  crop_level_join[is.na(crop_level_join$GCAM_commodity), ] -> na_crops
+
+  # This second link uses grepl to assign crops that are still NA to GCAM crops.
+  na_crops %>%
```

Comment:
You've nailed it.
Couple of small suggestions:
1. You won't need the `dplyr::` call for `case_when` and `rename` (keep it for select though, due to potential conflict with the `raster` package. If these aren't working then it's because you need to document the package. If having trouble with this then let's set up a call to resolve.
2. Everything with "Dbl Crop" is currently assigned to `MiscCrop`. Do these cases specify the two crops? If so, it might be more sensible to select the first one named.
3. Take a new line for each case (e.g., line 328 should be two lines)